satellite/marketingweb/form.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package marketingweb
+
+import (
+	"net/http"
+	"reflect"
+	"strconv"
+	"time"
+
+	"github.com/gorilla/schema"
+
+	"storj.io/storj/internal/currency"
+	"storj.io/storj/satellite/rewards"
+)
+
+// parseOfferForm decodes POST form data into a new offer.
+func parseOfferForm(w http.ResponseWriter, req *http.Request) (o rewards.NewOffer, e error) {
```

Comment:
nit -- move this down below var and init declarations

Reply:
parseOfferForm has higher importance than the var and init here.

Reply:
? could you explain. This is new to me.

Reply:
There are roughly few ideas for ordering package:

1. coherence
2. convention
3. importance
4. (I may have forgotten something)

Coherence says, that try to keep similar things close together. e.g. methods for the same type, this of course can be flipped when there is a large set of types with similar methods, that keep similar methods together. Either way the principle is about "grouping and keeping similar things close to each other".

Convention says, that order things like everything else. e.g. type definition is usually followed by `New` or other constructors, which is followed by `Close` (partly because of convention, partly because it is easier to see what to close, i.e. coherence).

Importance says that, things that matter less go to the bottom. This is about the principle of what you want people to read when they open the file. In this case the important concept is about parsing the form and then comes how it's actually implemented.

Reply:
Makes sense, I have never seen this done and thought I remembered reading the convention was to group the variable initialization and  init functions at the top of the file since that was what the compiler initialized first

Reply:
Yeah, the "initialization" is the convention part... neither is wrong. Here it seemed the init wasn't that important, hence to the bottom it went.

Reply:
Thanks for explaining!